Minor edits in previous blog, add new blog for Item 10

krshrimali opened the pull request:
Changes as per title.

Original link: https://krshrimali.github.io/Prefer-Scoped-Enums-Over-Unscoped-Enums/

cc: @kshitij12345 @khushi-411

krshrimali committed:
Minor edits in previous blog, add new blog for Item 10

kshitij12345 commented on _posts/2021-08-14-Prefer-Scoped-Enums-Over-Unscoped-Enums.md:
````diff
@@ -0,0 +1,210 @@
+
+```cpp
+// Assume Color is declared like above
+Color c = red; // valid since Enumerator white is leaded to the scope Color is in
````
What does `is leaded to the scope Color is in` mean?

krshrimali replied:
Ah, typo. Should be _leaked_, does that make sense?

kshitij12345 commented on _posts/2021-08-14-Prefer-Scoped-Enums-Over-Unscoped-Enums.md:
````diff
@@ -0,0 +1,210 @@
+}
+```
+
+The C++-9 Style Enums are termed as uncoped Enums (because of leaking names).
````
C++-98? Unscoped?

krshrimali replied:
Typo again, sorry. Thanks for pointing.

kshitij12345 commented on _posts/2021-08-14-Prefer-Scoped-Enums-Over-Unscoped-Enums.md:
```diff
@@ -0,0 +1,210 @@
+
+**C++98:**
+
+* Unscoped enums can not be forward-declared.
```
Can be with hacks? As you have mentioned below? 
I might be wrong though.

krshrimali replied:
Not in C++-98, the compiler needs to know the values in the enumerator to know the storage of the enum.

I'll add these to good reads:

* Proposal for forward declaration to enums (accepted), dated 2008: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2008/n2764.pdf
* Forward Declaring an Enum in C++? https://stackoverflow.com/questions/71416/forward-declaring-an-enum-in-c

kshitij12345 replied:
Ah. I see. Thanks for the resources!

kshitij12345 reviewed (approved):
LGTM! Overall looks good! Few minor comments to be updated.

Thanks! @krshrimali

krshrimali committed:
Typos fixed, added acknowledgement, and good reads

krshrimali committed:
Fix link

krshrimali commented:
Added acknowledgement, fixed typos and added the 2 links suggested before to good reads.

@kshitij12345 - Thanks a lot for your reviews, they are really helpful. Merging this now. 🎉
